cypress/support/MonitorDocumentChanges.js

```diff
@@ -68,31 +68,34 @@ var MonitorDocumentChanges = (function () {
       })
     })
 
-    Cypress.Commands.add('DoesNotContain', { prevSubject: 'optional' }, (subject, arg1, arg2) => {
-      helpers.ConLog(`cy.DoesNotContain(${subject}, ${arg1}, ${arg2})`, `Start - Last DOM change was ${MillisecondsSinceLastChange()} milliseconds ago`)
+    // The last "expectFailure" parameter is used only for testing that this function works.
+    // To use it set up the conditions where the selector (and optional text if appropriate)
+    // does exist on the page and call this with "expectFailure" set to true. If this finds
+    // the selector then this method will return true, otherwise false.
+    // Without setting "expectFailure" this method will throw an exception on failure.
+    Cypress.Commands.add('DoesNotContain', { prevSubject: 'optional' }, (subject, selector, textItShouldNotContain, expectFailure = false) => {
+      let functionSignature = `cy.DoesNotContain(${subject}, ${selector}, ${textItShouldNotContain})`
+      helpers.ConLog(functionSignature, `Start - Last DOM change was ${MillisecondsSinceLastChange()} milliseconds ago`)
       cy.wrap(700, { timeout: 60000 }).should('lte', 'MillisecondsSinceLastChange').then(() => {
-        helpers.ConLog(`cy.DoesNotContain()`, `DOM Is Stable`)
-
-        let selector
-        if (arg2) selector = arg2
-        else selector = arg1
+        helpers.ConLog(functionSignature, `DOM Is Stable`)
 
         let elements
         if (subject) elements = Cypress.$(subject).find(selector)
         else elements = Cypress.$(selector)
 
-        helpers.ConLog(`cy.DoesNotContain()`, `Found ${elements.length} for selector: ${selector}`)
+        helpers.ConLog(functionSignature, `Found ${elements.length} for selector: ${selector}`)
 
-        if (elements.length > 0) {
-          if (arg2) {
-            elements = Cypress.$(elements).find(`:contains(${arg2})`)
-            helpers.ConLog(`cy.DoesNotContain()`, `Found ${elements.length} containing text: ${arg2}`)
-          }
+        if (elements.length > 0 && textItShouldNotContain) {
+          elements = Cypress.$(elements).find(`:contains(${textItShouldNotContain})`)
+          helpers.ConLog(functionSignature, `Found ${elements.length} containing text: ${textItShouldNotContain}`)
+        }
 
-          if (elements.length > 0)
-            throw `Selector "${arg1}" + "${arg2}" was expected to be missing from the DOM, instead we found ${elements.length} instances of it.`
+        if (elements.length > 0) {
+          if (expectFailure) return true;
+          throw `selector: "${selector}" & textItShouldNotContain: "${textItShouldNotContain}" was expected to be missing from the DOM, instead we found ${elements.length} instances of it.`
         }
-        helpers.ConLog(`cy.DoesNotContain()`, `PASSED - Selector: ${selector} was NOT Found as Expected`)
+        helpers.ConLog(functionSignature, `PASSED - Selector was NOT Found as Expected`)
+        if (expectFailure) return false;
       })
     })
```

Comment:
If elements.length evaluates to zero length, and expectFailure is false (default), the code's return value is non-deterministic. Suggest considering a default initializer to textItShouldNotContain

Reply:
> Suggest considering a default initializer to textItShouldNotContain

Can you give me a reason why. My only resistance is that I don't see the need so less code is less noise.

Reply:
> If elements.length evaluates to zero length, and expectFailure is false (default), the code's return value is non-deterministic.

This is intentional.

When `expectFailure` is true, that is test mode. I need the return value to test it. I cannot use try/catch since Cypress catches all exceptions and stops the test.

When `expectFailure` is false, that is the normal run mode. I need it to return the value that came from the optional previous cy command that this may be chained off of in order to do further work. So we could code something like this for example:

    `cy.Get('.selector').DoesNotContain('.notSelector').Click()`

So in that example, the DoesNotContain would be an intermediate validation step in the chain, and the click would click on the first element returned from the Get(`.selector`).